misc/bstr.h

```diff
@@ -80,6 +80,10 @@ double bstrtod(struct bstr str, struct bstr *rest);
 void bstr_lower(struct bstr str);
 int bstr_sscanf(struct bstr str, const char *format, ...);
 
+// Decode a string containing hexadecimal data. All non-hex characters will
+// be silently ignored, as will a lone trailing digit.
+struct bstr bstr_decode_hex(void *talloc_ctx, struct bstr hex);
```

Comment:
You mean ignoring only any of '\r', '\n', '\t' and ' '? Could be done, if it makes you much happier